postproengine/spod.py

```diff
@@ -0,0 +1,685 @@
+# Get the location where this script is being run
+import sys, os
+scriptpath = os.path.dirname(os.path.realpath(__file__))
+basepath   = os.path.dirname(scriptpath)
+# Add any possible locations of amr-wind-frontend here
+amrwindfedirs = ['../',
+                 basepath]
+for x in amrwindfedirs: sys.path.insert(1, x)
+
+from postproengine import registerplugin, mergedicts, registeraction
+import postproamrwindsample_xarray as ppsamplexr
+import postproamrwindsample as ppsample
+import pickle
+import matplotlib.pyplot as plt
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+
+import pandas as pd
+import numpy as np
+import scipy as sp
+import matplotlib
+import matplotlib.pyplot as plt
+import amrwind_frontend as amrwind
+from itertools import product
+
+def plot_radial(Ur,theta,r,rfact=1.4,cmap='coolwarm',newfig=True,vmin=None,vmax=None,ax=None):
+    if newfig == True:
+        fig, ax = plt.subplots(subplot_kw={'projection':'polar'})
+    else:
+        ax = ax
+
+    LR = r[-1]
+    #ax=plt.subplot(111,polar=True)
+    #im = ax.pcolormesh(theta,r,Ur,cmap='coolwarm',vmin=0,vmax=8)
+    if vmin == None or vmax == None:
+        im = ax.pcolormesh(theta,r,Ur,cmap=cmap)
+    else:
+        im = ax.pcolormesh(theta,r,Ur,cmap=cmap,vmin=vmin,vmax=vmax)
+    #im = ax.pcolormesh(theta,r,Ur,cmap='jet')
+    #cbar = plt.colorbar(im,orientation='horizontal')
+    #cbar = plt.colorbar(im,orientation='vertical')
+
+    # ---- mod here ---- #
+    #ax.set_theta_zero_location("N")  # theta=0 at the top
+    #ax.set_theta_direction(-1)  # theta increasing clockwise
+    ax.set_rmax(LR)
+    #ax.set_yticks([0, LR/4, LR/2, 3*LR/4,LR])  # less radial ticks
+    ax.set_yticks([0,LR/1.4, LR])  # less radial ticks
+    #ax.set_rlabel_position(-22.5)  # get radial labels away from plotted line
+    ax.set_yticklabels(["$0$","$R$","1.4$R$"])
+    ax.grid(True)
+    # skoet the locations of the angular gridlines
+    #lines, labels = thetagrids(range(45, 360, 90))
+
+    # set the locations and labels of the angular gridlines
+    lines, labels = plt.thetagrids([0,45,90,135,180,225,270,315], ("$90^\circ$","$45^\circ$","$0^\circ$","$315^\circ$","$270^\circ$","$225^\circ$","$180^\circ$","$135^\circ$"))
+    #ax.spines['polar'].set_visible(False)
+    return im
+
+
+def read_cart_data(ncfile,group,trange,iplanes):
```

Comment:
I keep repeating several lines of code in this function across executors and I'm hoping we can move most of it to the postproamrwindsample_xarray.py. 

First, perhaps the choice between getPlaneXR and getFullPlaneXR can happen inside getPlaneXR to avoid repeating these if statement. This also allows us to take advantage of reading multiple files with the current getPlaneXR, while punting the equivalent getFullPlaneXR functionality until we have time for it. 

Second, there's a lot of code below around slicing the data to agnostic to the choice of x,y, and z ordering in the pre-processing setup. Perhaps this should also be moved to getPlaneXR with an option to return the data as a numpy array instead of as a dictionary.

Reply:
I agree with that, we should merge getPlaneXR and getFullPlaneXR into a single function to streamline things.  We should also talk about getting some common code to work around the choice of x/y/z coordinates -- I'm working on something now to deal with that.